src/sqlfluff/core/rules/std.py

```diff
@@ -3061,7 +3061,11 @@ def _lint_aliases_in_join(
             table_ref = table_exp.get_child("object_reference")
 
             # If this is self-join - skip it
-            if base_table.raw == table_ref.raw and base_table != table_ref:
+            if (
+                base_table
```

Comment:
We weren't checking if there actually was a base_table before taking its `raw`. Same below with `used_alias_ref`, get_child was returning a `None` and then failing when we tried to get the `raw`.